src/editor.cpp

```diff
@@ -1967,8 +1967,8 @@ void Editor::objectsView_onMousePress(QMouseEvent *event) {
     if (map_item && map_item->paintingMode != MapPixmapItem::PaintMode::EventObjects) {
         return;
     }
-    if (this->map_edit_mode == "paint" && event->buttons() & Qt::RightButton) {
-        this->map_edit_mode = "select";
+    if (this->obj_edit_mode == "paint" && event->buttons() & Qt::RightButton) {
+        this->obj_edit_mode = "select";
```

Comment:
I think this function originally checking and setting `map_edit_mode` when editing objects may have been a mistake, which would cause the default tool to be the `Pointer` when switching back to the `Map` tab. If this was intentional then let me know.